### PR TITLE
[backport/1.21] deps: Bump `com_github_wasmtime` -> 1.0.2 (#24086)

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@PiotrSikora @RiverPhillips this fixes an incoming CVE - the patch applies pretty cleanly so i think its ok to take it

@mattklein123 as this is a bit of release blocker i will hold off pushing the tag button until this has landed

**Review by mattklein123 — APPROVED** [review]

File: bazel/repository_locations.bzl
Comment (PiotrSikora) [inline bazel/repository_locations.bzl:770]: Again, this only updates the Wasmtime C API (i.e. the glue code), the core of Wasmtime is what Proxy-Wasm C++ Host points at, which is still v0.39.1 in case of commit in this PR.

Comment (phlax) [reply bazel/repository_locations.bzl:770]: arggh - i thought i had updated proxy-wasm-c++-host ill check this again ...

Comment (PiotrSikora) [reply bazel/repository_locations.bzl:770]: There is no v1.0.2 in Proxy-Wasm C++ Host, we have v1.0.1 and v2.0.0 at the HEAD.

Comment (phlax) [reply bazel/repository_locations.bzl:770]: k - i see - can we update there - i think the change from 1.0.1 is just the CVE fix/es

Comment (PiotrSikora) [reply bazel/repository_locations.bzl:770]: Considering that Wasmtime was already updated in the stable release, it would be easier to go straight to v2.0.2, since we don't maintain old branches in Proxy-Wasm C++ Host, and HEAD is already at v2.0.0. Could you send a PR?

Comment (phlax) [reply bazel/repository_locations.bzl:770]: i can create a PR but `master` is beyond this now - im wondering can we create a `wasmtime-1.0.2` branch ?

Comment (phlax) [reply bazel/repository_locations.bzl:770]: k - yep - ill update -> 2.0.2

**Pull request merged**
